src/profile_manager/views.py

```diff
@@ -103,6 +104,27 @@ def get_queryset(self):
         return self.queryset_append(profiles_qs)
 
 
+@method_decorator(staff_member_required, name='dispatch')
+class ProfileListBlock(ProfileList):
+    """lists all students in a given block, is accessed through the block list view and acts as a hybrid profile list and block detail view"""
+    view_type = ViewTypes.BLOCK
+
+    def get_queryset(self):
+        """block object is queried via pk kwarg in request from block list view, then a queryset of profiles is generated via relatedmanager"""
+        block_pk = self.kwargs['pk']
+        block = Block.objects.filter(pk=block_pk).first()
```

Comment:
Don't filter.first() unless you are expecting the possibility of the block not existing (which would result in an empty queryset).  Instead you should use `.get(pk=block_pk)`

To handle possibility of manually entered url with a block pk that doesn't exist, django has a convenience function for that which is probably what you should use here: `get_object_or_404`.  Just make sure the 404 is properly raised (since this is in the get_queryset method I don't know if it works here):

https://docs.djangoproject.com/en/3.2/topics/http/shortcuts/#get-object-or-404

whether you want a test for the bad pk raising a 404 is up to you (what does it currently do if you manually enter 999 into the url?)

Reply:
get_object_or_404 function implemented, added an additional line in the page status codes test to catch new 404 conditional.